sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java

```diff
@@ -1029,6 +1033,7 @@ public RxDocumentServiceRequest clone() {
         rxDocumentServiceRequest.forcePartitionKeyRangeRefresh = this.forcePartitionKeyRangeRefresh;
         rxDocumentServiceRequest.useGatewayMode = this.useGatewayMode;
         rxDocumentServiceRequest.requestContext = this.requestContext;
+        rxDocumentServiceRequest.faultInjectionRequestContext = new FaultInjectionRequestContext(this.faultInjectionRequestContext);
```

Comment:
IMO worth adding a comment here - that this ctor will ensure hitcounting still works - Took me a while to understand taht this ctor is doing a hybrid of keeping hitcountMap but creating new transportrequestIdbased map

Reply:
In three months I probably would need even longer to get what is happening here.

Reply:
haha, make sense, added comments